### PR TITLE
enhance: Use arrow cdata to avoid proto marshaling delete data

### PR DESCRIPTION
**Pull request by congqixia**
Related to #35303

Reference https://arrow.apache.org/docs/format/CDataInterface.html

### CODE REVIEW

File: internal/core/src/common/type_c.h [hunk 1]
```diff
@@ -126,6 +126,53 @@ typedef struct CNewSegmentResult {
...
 } CNewSegmentResult;
+
+#ifndef ARROW_C_DATA_INTERFACE
+#define ARROW_C_DATA_INTERFACE
+
+#define ARROW_FLAG_DICTIONARY_ORDERED 1
+#define ARROW_FLAG_NULLABLE 2
+#define ARROW_FLAG_MAP_KEYS_SORTED 4
+
+struct ArrowSchema {
```
Comment (tedxu) [inline internal/core/src/common/type_c.h:137]: These structures are already defined in `arrow/c/abi.h`, we should include that file instead.

File: internal/querynodev2/segments/segment.go [hunk 2]
```diff
@@ -785,48 +789,43 @@ func (s *LocalSegment) Delete(ctx context.Context, primaryKeys []storage.Primary
...
 
-	dataBlob, err := proto.Marshal(ids)
-	if err != nil {
-		return fmt.Errorf("failed to marshal ids: %s", err)
-	}
+	defer pkArr.Release()
+
+	var cschema cdata.CArrowSchema
+	var carr cdata.CArrowArray
+	cdata.ExportArrowArray(pkArr, &carr, &cschema)
```
Comment (tedxu) [inline internal/querynodev2/segments/segment.go:822]: The arrays and schemas should get released somewhere.

File: internal/core/src/common/type_c.h (hunk 1 above)
Comment (congqixia) [reply internal/core/src/common/type_c.h]: done

File: internal/querynodev2/segments/segment.go (hunk 2 above)
Comment (congqixia) [reply internal/querynodev2/segments/segment.go:822]: fixed

**Comment by shaoting-huang** [comment]
Can we get the overhead from CGO?

**Comment by congqixia** [comment]
rerun go-sdk

**Comment by congqixia** [comment]
rerun go-sdk

**Comment by congqixia** [comment]
rerun go-sdk

**Comment by congqixia** [comment]
rerun go-sdk

**Pull request closed without merging**